src/braket/circuits/circuit.py

```diff
@@ -1030,6 +1029,12 @@ def _flatten(addable):
 
         return self
 
+    def adjoint(self) -> Circuit:
+        circ = Circuit([instr.adjoint() for instr in reversed(self.instructions)])
+        for result_type in self._result_types:
+            circ.add_result_type(result_type)
```

Comment:
I always worry that there's data we're forgetting to copy over. For example, do we need to copy over the parameters?

Reply:
Nope, that's included in the gates. Right now, there are only two things you can add to a circuit: instructions and result types (and other circuits, which are made up of these)